lib/Sema/ConstraintSystem.cpp

```diff
@@ -4427,6 +4427,71 @@ static bool diagnoseAmbiguityWithContextualType(
   return true;
 }
 
+/// Diagnose problems with generic requirement fixes that are anchored on
+/// one callee location. The list could contain different kinds of fixes
+/// i.e. missing protocol conformances at different positions,
+/// same-type requirement mismatches, etc.
+static bool diagnoseAmbiguityWithGenericRequirements(
+    ConstraintSystem &cs,
+    ArrayRef<std::pair<const Solution *, const ConstraintFix *>> aggregate,
+    ConstraintLocator *calleeLocator) {
+  // If all of the fixes point to the same overload choice,
+  // we can diagnose this an a single error.
+  bool hasNonDeclOverloads = false;
+
+  llvm::SmallSet<ValueDecl *, 4> overloadChoices;
+  for (const auto &entry : aggregate) {
+    if (auto overload =
+            entry.first->getOverloadChoiceIfAvailable(calleeLocator)) {
+      if (auto *D = overload->choice.getDeclOrNull()) {
+        overloadChoices.insert(D);
+      } else {
+        hasNonDeclOverloads = true;
+      }
+    }
+  }
+
+  {
+    auto &primaryFix = aggregate.front();
+
+    if (overloadChoices.size() > 0) {
+      // Some of the choices are non-declaration,
+      // let's deligate that to ambiguity diagnostics.
```

Comment:
Typo deligate -> delegate?

Reply:
Nevermind, I just didn't know [deligate](https://www.thefreedictionary.com/Deligate) was a word. Although the context delegate also makes sense :)

Reply:
I will switch it, no problem :) I’d like to make some changes to address the FIXME anyway

Reply:
✅